apps/omg_watcher_info/test/omg_watcher_info/exit_consumer_test.exs

```diff
@@ -0,0 +1,162 @@
+# Copyright 2019-2020 OmiseGO Pte Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+defmodule OMG.WatcherInfo.ExitConsumerTest do
+  use ExUnitFixtures
+  use ExUnit.Case, async: false
+  use OMG.Fixtures
+
+  alias OMG.Crypto
+  alias OMG.Utxo
+  alias OMG.Utxo.Position
+  alias OMG.WatcherInfo.DB
+  alias OMG.WatcherInfo.ExitConsumer
+
+  require Utxo
+
+  setup_all do
+    {:ok, _} =
+      GenServer.start_link(
+        ExitConsumer,
+        [topic: :watcher_test_topic, bus_module: __MODULE__.FakeBus],
+        name: TestExitConsumer
+      )
+
+    _ =
+      on_exit(fn ->
+        with pid when is_pid(pid) <- GenServer.whereis(TestExitConsumer) do
```

Comment:
Couldn't we use pid returned by `GenServer.start_link` in line 29 ?

Reply:
The only thing here is the process could be terminated at this point, as it’s linked to the test process. So the code makes sure it’s still alive before trying to stop it.